lib/internal/bootstrap/node.js

```diff
@@ -625,5 +627,59 @@
     new vm.Script(source, { displayErrors: true, filename });
   }
 
+  function setupAllowedFlags() {
+    // This builds process.allowedNodeEnvironmentFlags
+    // from data in the config binding
+
+    const replaceDashesRegex = /-/g;
+    const leadingDashesRegex = /^--?/;
+    const trailingValuesRegex = /=.*$/;
+
+    const trimLeadingDashes = (flag) => flag.replace(leadingDashesRegex, '');
+
+    const {
+      allowedV8EnvironmentFlags,
+      allowedNodeEnvironmentFlags
+    } = process.binding('config');
+
+    // for comparison purposes only
+    const v8Flags = new Set(allowedV8EnvironmentFlags
+      .map(trimLeadingDashes));
+    const nodeFlags = new Set(allowedNodeEnvironmentFlags
+      .map(trimLeadingDashes));
+
+    class NodeEnvironmentFlagsSet extends Set {
+      constructor(...args) {
+        // There is no way to actually freeze or otherwise
+        // make a Set immutable, AFAIK.  Consider this
+        // "best effort" using "minimal effort"
+        super(...args);
+
+        this.add = () => this;
+        this.clear = () => {};
+        this.delete = () => false;
+        Object.freeze(this);
+      }
+
+      has(key) {
+        if (typeof key === 'string') {
+          key = key.replace(trailingValuesRegex, '');
```

Comment:
You probably want to be robust here against deletion or replacement of string replace, to avoid exposing this RegExp (which would increase the size of the observable api)

Reply:
not entirely sure what this looks like. grab a ref to `Strong.prototype.replace` and use that?

Reply:
`const replace = Function.call.bind(String.prototype.replace);` and then `replace(key, trailingValuesRegex, '');`